homeassistant/components/mqtt/sensor.py

```diff
@@ -284,12 +284,23 @@ def _update_state(msg: ReceiveMessage) -> None:
             if self.device_class is None:
                 self._attr_native_value = new_value
                 return
-            if (payload_datetime := dt_util.parse_datetime(new_value)) is None:
+            try:
+                if (payload_datetime := dt_util.parse_datetime(new_value)) is None:
+                    _LOGGER.warning(
+                        "Invalid state message '%s' from '%s'", msg.payload, msg.topic
+                    )
+                    self._attr_native_value = None
+                    return
+            except ValueError as value_error:
                 _LOGGER.warning(
-                    "Invalid state message '%s' from '%s'", msg.payload, msg.topic
+                    "Invalid state message '%s' from '%s': %s",
+                    msg.payload,
+                    msg.topic,
+                    value_error.args,
```

Comment:
Let's simplify this and do it in the same way as when we handle last reset timestamps just below, we don't care about what's in the exception, it's enough to log the offending message.